crates/bevy_ecs/src/change_detection.rs

```diff
@@ -444,6 +444,18 @@ impl<'a> MutUntyped<'a> {
     pub fn as_ref(&self) -> Ptr<'_> {
         self.value.as_ref()
     }
+
+    /// Transforms this [`MutUntyped`] into a [`Mut<T>`] with the same lifetime.
+    ///
+    /// # Safety
+    ///
+    /// Must point to a valid `T`.
```

Comment:
This should also mentioned that the pointer within must be properly aligned, as required of `&mut T`.